### PR TITLE
🕷️ 1077 Fix Resolver Auth

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

Adds reusable authorization hook for checking GQL Query permissions
Adds hook to validateActiveSubmissionData, clearFileDataFromActiveSubmission, commitActiveSubmissionData 
Move mutations into their own folder

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/gh/icgc-argo/argo-clinical/1077

**Type of Change**

- [x] Bug
- [x] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/decorators/index.ts [hunk 1]
```diff
@@ -162,6 +162,22 @@ export function HasProgramWriteAccess(programIdExtractor: Function) {
...
+
+  const scopes = decodedToken.scopes;
+  const requiredScopes = [`PROGRAMDATA-${programId}.WRITE`, 'CLINICALSERVICE.WRITE'];
+  const hasWriteAccess = hasScope(requiredScopes, scopes);
+
+  if (!hasWriteAccess) throw new Error('Query does not have the required permissions');
+
+  return hasWriteAccess;
+}
+
```
Comment (demariadaniel) [inline src/decorators/index.ts:180]: Would preferably update this file to use a HOC/decorator pattern
If `!hasWriteAccess` we throw an error, if `hasWriteAccess` you call the child function

File: src/routes/submission.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ router.post(
   wrapAsync(submissionAPI.uploadClinicalTsvFiles),
 );
 
-router.post('/validate/:versionId', wrapAsync(submissionAPI.validateActiveSubmission));
+router.post('/validate/:versionId', wrapAsync(submissionAPI.validateActiveSubmissionRequest));
 router.post('/commit/:versionId', wrapAsync(submissionAPI.commitActiveSubmission));
```
Comment (demariadaniel) [inline src/routes/submission.ts:38]: Separate Request Handler methods from GQL Resolvers

File: src/schemas/clinical-mutations/clearClinicalSubmission.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (demariadaniel) [inline src/schemas/clinical-mutations/clearClinicalSubmission.ts:1]: Move mutations to a separate folder

**Comment by UmmulkiramR** [comment]
Looks good. Thanks.

**Review by UmmulkiramR — APPROVED** [review]

**Pull request merged**